File: 2570-merge-two-2d-arrays-by-summing-values/main.cpp
class Solution {
public:
    vector<vector<int>> mergeArrays(vector<vector<int>>& nums1, vector<vector<int>>& nums2) {
        
        unordered_map<int, int> _dict;

        for (const auto& nums: nums1) {
            _dict[nums[0]] = nums[1];
        }

        for (const auto& nums: nums2) {
            _dict[nums[0]] += nums[1];
        }

        vector<vector<int>> ans;
        for (const auto& [key, value]: _dict) {
            ans.push_back({key, value});
        }

        sort(ans.begin(), ans.end());
        return ans;
    }
};

File: 2570-merge-two-2d-arrays-by-summing-values/main.go
func mergeArrays(nums1 [][]int, nums2 [][]int) [][]int {
    
    _dict :=  make(map[int]int)

    for _, nums := range nums1 {
        _dict[nums[0]] = nums[1] 
    }

    for _, nums := range nums2 {
        _dict[nums[0]] += nums[1]
    }

    var ans [][]int 
    for key, value := range _dict {
        ans = append(ans, []int{key, value})
    }

    sort.Slice(ans, func(i, j int) bool {
        return ans[i][0] < ans[j][0]
    })

    return ans
}
